src/iq.rs

```diff
@@ -484,4 +488,110 @@ mod tests {
+            let client = IQClient::new(
+                mock_server.parse().unwrap(),
+                "".to_string(),
+                "".to_string(),
+                "".to_string(),
+                public_app_id.to_string(),
+                0,
+            );
+            let internal_application_id = client
+                .get_internal_application_id(public_app_id.to_string())
+                .expect("Failed to retrieve application ID");
+
+            assert_eq!(internal_application_id.applications.len(), 1);
+            assert_eq!(
+                internal_application_id.applications[0].public_id,
+                public_app_id
+            );
+        }
+        mock.assert();
+    }
+
+    #[test]
+    fn test_get_internal_app_id_when_empty() {
+        let public_app_id = "iqPublicApplicationId".to_string();
+        let mock_path = format!("/api/v2/applications?publicId={}", public_app_id);
+
+        let raw_json: &str = r#"{
+                "applications": [
+                ]
+            }"#;
+        let mock = mock("GET", mock_path.as_str())
+            .with_header("CONTENT_TYPE", "application/json")
+            .with_body(raw_json)
+            .create();
+        {
+            let mock_server = &mockito::server_url();
+            let client = IQClient::new(
+                mock_server.parse().unwrap(),
+                "".to_string(),
+                "".to_string(),
+                "".to_string(),
+                public_app_id.to_string(),
+                0,
+            );
+            let internal_application_id = client
+                .get_internal_application_id(public_app_id.to_string())
+                .expect("Failed to retrieve application ID");
+
+            assert_eq!(internal_application_id.applications.len(), 0);
+        }
+        mock.assert();
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "empty application ID: GeneralError(\"could not get internal application id for public application id: "
+    )]
+    fn test_audit_with_iq_server_when_applications_empty() {
+        let public_app_id = "iqPublicApplicationId".to_string();
+        let mock_path = format!("/api/v2/applications?publicId={}", public_app_id);
+
+        let raw_json: &str = r#"{
+                "applications": [
+                ]
+            }"#;
+        let mock = mock("GET", mock_path.as_str())
+            .with_header("CONTENT_TYPE", "application/json")
+            .with_body(raw_json)
+            .create();
+        {
+            let mock_server = &mockito::server_url();
+            let client = IQClient::new(
+                mock_server.parse().unwrap(),
+                "".to_string(),
+                "".to_string(),
+                "".to_string(),
+                public_app_id.to_string(),
+                0,
+            );
+            let _internal_application_id = client
+                .audit_with_iq_server("".to_string())
+                .expect("empty application ID");
```

Comment:
This is fine as is, but as a style suggestion I'd probably use [`expect_err`](https://doc.rust-lang.org/std/result/enum.Result.html#method.expect_err) and avoid the panic via something like:

``` rust
let actual_error = client
    .audit_with_iq_server("".to_string())
    .expect_err("Expected to fail, but audit_with_iq_server call succeeded");
assert_eq!(format!("{}", actual_error), "empty application ID: GeneralError(\"could not get internal application id for public application id: ");
```

If not the most clean with the `format` macro call, but I think it's still a good approach in application cases where you don't want to create a whole error enum with `thiserror`.

Reply:
Much better. Done: c0d8248